cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/ProgramWorkflowAction.java

```diff
@@ -60,22 +61,22 @@ public void configure() {
 
   @Override
   public void run() {
+    String prettyProgramType = ProgramType.valueOf(programType.name()).getPrettyName();
     try {
       String programName = getContext().getSpecification().getProperties().get(PROGRAM_NAME);
       // this should not happen, since null is only passed in from WorkflowDriver, only when calling configure
       if (programWorkflowRunner == null) {
         throw new UnsupportedOperationException("Operation not allowed.");
       }
       Runnable programRunner = programWorkflowRunner.create(programName);
-      LOG.info("Starting Program for workflow action: {}", programName);
+      LOG.info("Starting {} Program '{}' in workflow", prettyProgramType, programName);
       programRunner.run();
 
       // TODO (terence) : Put something back to context.
 
-      LOG.info("{} Program {} workflow action completed",
-               programType != null ? programType.name() : null, programName);
+      LOG.info("{} Program '{}' in workflow completed", prettyProgramType, programName);
     } catch (Exception e) {
-      LOG.info("Failed to execute {} Program {} in workflow", programType, programName, e);
+      LOG.error("Failed to execute {} Program '{}' in workflow", prettyProgramType, programName, e);
```

Comment:
in general, shouldn't log an exception if it is being throw. I think we can just get rid of this `try-catch` block.